src/corehost/common/pal.h

```diff
@@ -50,16 +50,26 @@
 
 #endif
 
-
+// When running on a platform that is not supported in RID fallback graph (because it was unknown
+// at the time the SharedFX in question was built), we need to use a reasonable fallback RID to allow
+// consuming the native assets.
+//
+// For Windows and OSX, we will maintain the last highest RID-Platform we are known to support for them as the 
+// degree of compat across their respective releases is usually high.
+//
+// We cannot maintain the same (compat) invariant for linux and thus, we will fallback to using lowest RID-Plaform.
 #if defined(_WIN32)
 #define LIB_PREFIX
 #define MAKE_LIBNAME(NAME) (_X(NAME) _X(".dll"))
+#define FALLBACK_HOST_RID _X("win10")
```

Comment:
See the conversation here:  https://github.com/dotnet/core-setup/pull/1689#discussion_r105297258.

If we took that approach, we would miss a lot of assets and more than likely the app wouldn't work by default, just because a new macOS version came out.